### PR TITLE
Gat/mrd pipeline fixes

### PR DESCRIPTION
**Pull request by GatKrieger**
still work in progress, but would love to have a review.

### CODE REVIEW

File: ugvc/reports/mrd_automatic_data_analysis.ipynb [hunk 1]
```diff
@@ -279,6 +281,17 @@
...
+    "            \"C->A\",\n",
+    "            \"C->G\",\n",
+    "            \"C->T\",\n",
+    "            \"T->A\",\n",
+    "            \"T->C\",\n",
+    "            \"T->G\",\n",
+    "        ]\n",
+    "        for mut in all_muts:\n",
+    "            if mut not in x.index:\n",
+    "                x.loc[mut] = 0\n",
```
Comment (kenissur) [inline ugvc/reports/mrd_automatic_data_analysis.ipynb]: I think x.reindex(all_muts).fillna(0).astype(int) would save you the loop.

File: ugvc/mrd/mrd_utils.py [hunk 2]
```diff
@@ -176,11 +176,6 @@ def collect_coverage_per_locus_gatk(coverage_csv, df_sig):
     coverage_gatk = pd.read_csv(coverage_csv, usecols=["Chrom", "Pos", "Total_Depth"])
     coverage_gatk.rename(columns={"Chrom": "chrom", "Pos": "pos", "Total_Depth": "coverage"}, inplace=True)
     coverage_gatk.set_index(["chrom", "pos"], inplace=True)
-    if df_sig.index.names != coverage_gatk.index.names:
-        raise ValueError(
-            f"df_sig index names {df_sig.index.names} \
-            do not match coverage_gatk index names {coverage_gatk.index.names}"
-        )
```
Comment (kenissur) [inline ugvc/mrd/mrd_utils.py:183]: Why do you want to remove this check?

File: ugvc/reports/mrd_automatic_data_analysis.ipynb [hunk 3]
```diff
@@ -587,13 +577,20 @@
...
+    "    ).fillna(0).astype(int).drop(columns = [0]).groupby([\"signature_type\", \"signature\"]).sum()\n",
+    "\n",
     "    df_coverage = (\n",
     "        (df_signatures_in.groupby(\"signature\").agg({\"coverage\": \"sum\"}))\n",
     "        .fillna(0)\n",
     "        .astype(int)\n",
     "    )\n",
     "    df_tf = df_supporting_reads.join(df_coverage).fillna(0)\n",
     "    df_tf[\"corrected_denom\"] = df_tf[\"coverage\"] * denom_ratio\n",
+    "    df_tf[\"corrected_denom\"] = np.ceil(df_tf[\"corrected_denom\"])\n",
```
Comment (kenissur) [inline ugvc/reports/mrd_automatic_data_analysis.ipynb:593]: 1. Best to do this together with the line above
2. Why round this value?

File: ugvc/reports/mrd_automatic_data_analysis.ipynb [hunk 4]
```diff
@@ -716,31 +751,29 @@
...
    "outputs": [],
    "source": [
-    "df_tf_filt.to_csv(os.path.join(output_dir, basename + \".df_tf_filt_signature_filt_featuremap.csv\"))\n",
-    "df_tf_unfilt.to_csv(os.path.join(output_dir, basename + \".df_tf_unfilt_signature_filt_featuremap.csv\"))\n",
-    "df_tf_unfilt2.to_csv(os.path.join(output_dir, basename +  \".df_tf_filt_signature_unfilt_featuremap.csv\"))\n",
-    "\n",
-    "df_supporting_reads_per_locus_filt.to_csv(os.path.join(output_dir, basename + \".df_supporting_reads_per_locus_filt_signature_filt_featuremap.csv\"))\n",
-    "df_supporting_reads_per_locus_unfilt.to_csv(os.path.join(output_dir, basename +  \".df_supporting_reads_per_locus_unfilt_signature_filt_featuremap.csv\"))\n",
-    "df_supporting_reads_per_locus_unfilt2.to_csv(os.path.join(output_dir, basename + \".df_supporting_reads_per_locus_filt_signature_unfilt_featuremap.csv\"))"
+    "## Write tf tables to csv files\n",
```
Comment (kenissur) [inline ugvc/reports/mrd_automatic_data_analysis.ipynb]: * to an hdf file

**Review by kenissur — APPROVED** [review]
Just some minor comments

**Comment by kenissur** [comment]
@doron-st and @ilyasoifer, this is a major PR that updates the MRD pipeline. I would appreciate review on whatever you feel comfortable reviewing before we go ahead and merge.

File: ugvc/pipelines/mrd/sorter_stats_to_mean_coverage.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+    Parameters
+    ----------
+    sorter_stats_json : str
+        Path to Sorter statistics JSON file.
+    output_file : str
+        Path to output file.
+
+    """
+    mean_cvg = np.ceil(read_effective_coverage_from_sorter_json(sorter_stats_json)[0])
```
Comment (ilyasoifer) [inline ugvc/pipelines/mrd/sorter_stats_to_mean_coverage.py:41]: Could you give some clarification what effective_coverage means? Either here or in `read_effective_coverage_from_sorter_json`

**Review by ilyasoifer — APPROVED** [review]
Looks good, just minor comment.

File: ugvc/pipelines/mrd/sorter_stats_to_mean_coverage.py (hunk 5 above)
Comment (kenissur) [reply ugvc/pipelines/mrd/sorter_stats_to_mean_coverage.py:41]: In this case it's just the mean coverage, fixed, and clarified in the other function

**Pull request merged**